### DISCUSSION TITLE
Bird and bat detection and classification concurrently??

### DISCUSSION THREAD

**Discussion by dickestonia** [rdz-oss/BattyBirdNET-Pi#20] [Q&A]
Hi all,

I am currently awaiting parts to build out an outdoor box built on the ideas of some amazing people before me.

My question here is is there a way to incorporate both bird and bat analysis on one RPi or would I need 2 in the same box? Does BattybirdNET also still analyze daytime recordings for birds??

As a side note Python is a snake to me so I will say programming isn’t something I know anything about though I can Google flashing a Pi.

See these amazing builds here:

https://becausebirds.com/fun-raspberry-pi-project-identify-backyard-birds-24-7/?unapproved=32636&moderation-hash=e94eae198776d7ad85d1f24bade511e5#comment-32636

https://pixcams.com/building-a-birdnet-pi-real-time-acoustic-bird-id-station/

**Comment by arne1921KF** [comment]
If you look at [this enhancement idea under Issues](https://github.com/rdz-oss/BattyBirdNET-Pi/issues/5), you'll find that @rdz-oss has this in mind. 

I speak R, so Python is not a snake to me - but my Parseltongue is so weak I definitly can't pull this off. But this would be EXACTLY what I want. I just connected a WA EM2Pro to my Pi, which ran Birdnet-Pi in the flavour of @Nachtzuster and got a new SD for that purpose. The first microphone is still connected, so switching microphones - and models - would be just a great hack. Maybe this could be done with the dusk till dawn timer?

**Comment by rdz-oss** [comment]
That is now working. Currently can be tested in branch bb. Birds during day an bats during night. 

https://github.com/rdz-oss/BattyBirdNET-Pi/tree/bb

Working fine on my two test machines, including automatic switch with sunset/sundown derived from lt/lon.

curl -s https://raw.githubusercontent.com/rdz-oss/BattyBirdNET-Pi/bb/newinstaller.sh | bash

Waiting a bit and will then add it to main branch in a few days. More features to come.

Hope you enjoy the feature,

BR
Richard 

<img width="904" alt="Screenshot 2024-04-30 at 21 23 46" src="https://github.com/rdz-oss/BattyBirdNET-Pi/assets/63958211/b5eb606c-428e-4f9a-a119-899898fc8fdb">

**Reply by dickestonia** [reply]
Thank you so much Richard!

I am still awaiting a few parts and pieces for my build but am anxiously looking forward to it. 

This will be a very welcome surprise for my wife!

Rick

**Reply by rdz-oss** [reply]
You will need an ultrasound mic such as the audiomoth usb version with case or a wildlife acoustics echo meter touch 2. No USB sound card required, just a connector cable.  Pi 4 or 5 both work. Pi 5 is faster, but Pi 4 can use passive cooling. Birds are also recorded including their ultrasound parts that do not show up on other systems.

Enjoy, best regards,
Richard

**Reply by dickestonia** [reply]
That’s great news.

I just received the WA echo meter pro, might be overkill for a backyard enthusiast but somewhere I saw some comments about less signal noise with the pro version.

I did get a Pi 4B with cooler so that should be okay.

Is there the ability to run different microphones (one for day and one for night?) or would the WA echo meter pro be sufficient for birds during the daytime??

Thank you so much again for this!

rick

**Reply by rdz-oss** [reply]
Glad you enjoy the system Rick!
One mic will be enough. Both audiomoth and EMT2 also record in 'normal' sound and the system takes care of the rest. You will also see the ultrasound part of bird recordings - that is different from BirdNET system that cap at 48kHz. European blackbirds easily go into 30+kHz range for example. Robins are also champions at that. You can check out the system on my station running an audiomoth mic here: https://battybirdnet.duckdns.org/ on 384kHz. You can set it to 256kHz if you want to see more detail for birds, too.

Cheers,
Richard
